EIPS/eip-7761.md

```diff
@@ -62,46 +76,57 @@ There have been other solutions proposed to alleviate the problems related to la
-    
-In parallel to the ERC-721 / ERC-1155 problem, another potential risk has been brought to attention. Since EOFv1 prohibits `EXTDELEGATECALL` targetting legacy contracts, there exists a scenario where an EOF proxy contract accidentally upgrades its implementation to a legacy EVM one. Since reverting this or upgrading again (using current proxy standards) requires the implementation contract to be called, it would effectively render the contract unusable.
+### Including safeguarding against proxy bricking
     
-One potential solution to this would be to have a generalized `CONTRACT_KIND` instruction instead of `HASCODE` which would further discriminate the account into legacy or EOFv1 with a `0` / `1` / `2` return value, thereby providing means for an additional safeguard against such a scenario.
+In parallel to the ERC-721 / ERC-1155 problem, another potential risk has been brought to attention. Since EOFv1 prohibits `EXTDELEGATECALL` targeting legacy contracts, there exists a scenario where an EOF proxy contract accidentally upgrades its implementation to a legacy EVM one. Since reverting this or upgrading again (using current proxy standards) requires the implementation contract to be called, it would effectively render the contract unusable.
     
-This problem is potentially solvable on the application layer (proxy upgrade would include a direct check of the implementation contract being `EXTDELEGATECALL`-able) or even by re-enabling `EXTDELEGATECALL` to a legacy target in a future EVM upgrade.
+For this reason it was decided to have a generalized `EXTCODETYPE` instruction instead of a `HASCODE` instruction. This providing toe means to safeguard against such a scenario that a simpler instruction could not.
+
+### Relation to [EIP-7702] "Set EOA account code"
+
+After [EIP-7702] is activated, the discrimination between EOAs and contract accounts using `EXTCODESIZE` (or `ACCOUNT`) has an edge case: Whenever an EOA sets its code to a contract account which does not respond as expected to an `onERC721Received` (`onERC1155Received`) callback, transfers to it will revert, despite the recipient being able to interact with the token. This has been deemed unlikely to be a problem, as for the intended real-world uses of EIP-7702, those callbacks will be implemented by designator codes.
+
+Because of the requirement that EOF proxy contracts be able to determine if the new target is safe we cannot return a value for indicating that the account itself is delegated without also providing the delegated address. Instead, we return  the code for the account that has the delegated code, applying the delegation. This is because the executable code is what matters for proxy updates.
+
+This differs from EXTCODECOPY and EXTCODEHASH behavior which do return a hash and short code indicating the proxy. The behavior of EXTCODETYPE is closer to a "code reading operation" in purpose as it is meant to describe the execution behavior of the account.
```

Comment:
I think you meant "executing" here? Executing operations follow delegation (as EXTCODETYPE), reading operations read the delegation designator.
```suggestion
This differs from `EXTCODECOPY` and `EXTCODEHASH` behavior which do return a hash and short code indicating the proxy. The behavior of `EXTCODETYPE` is closer to a "code executing operation" in purpose as it is meant to describe the execution behavior of the account.
```